### PR TITLE
Addon pipeline to collect pygments symbols

### PR DESCRIPTION
**Pull request by keshav-space**
Refrence: https://github.com/nexB/purldb/issues/399

### CODE REVIEW

File: scanpipe/pipelines/collect_pygments_symbols.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+class CollectPygmentsSymbolsAndStrings(Pipeline):
+    """
+    Collect codebase symbols using pygments and keep them in extra data field.
+
+    Also collect strings and comments.
+    """
+
+    download_inputs = False
+    is_addon = False
```
Comment (pombredanne) [inline scanpipe/pipelines/collect_pygments_symbols.py:35]: should this be True?

**Review by pombredanne — CHANGES_REQUESTED** [review]
Looking fine ... though make this a n add-on pipeline and add an end to end test to run the pipeline too.

File: scanpipe/pipelines/collect_pygments_symbols.py (hunk 1 above)
Comment (keshav-space) [reply scanpipe/pipelines/collect_pygments_symbols.py]: Fixed, that was typo

**Review by tdruez — CHANGES_REQUESTED** [review]
@keshav-space See my comment.

Also, @pombredanne , do we  actually need 3 pipelines for symbol/string collection?
```
collect_pygments_symbols = scanpipe.pipelines.collect_pygments_symbols:CollectPygmentsSymbolsAndStrings
collect_source_strings = scanpipe.pipelines.collect_source_strings:CollectSourceStrings
collect_symbols = scanpipe.pipelines.collect_symbols:CollectSymbols
```

File: scanpipe/pipes/symbols.py [hunk 2]
```diff
@@ -65,3 +66,36 @@ def _collect_and_store_resource_symbols(resource):
...
+
+def _collect_and_store_pygments_symbols_and_strings(resource):
+    """
+    Collect symbols, strings and comments from a resource using pygments and store
+    them in the extra data field.
+    """
+    result = symbols_pygments.get_pygments_symbols(resource.location)
+    resource.update_extra_data({"source_symbols": result.get("source_symbols")})
+    resource.update_extra_data({"source_strings": result.get("source_strings")})
+    resource.update_extra_data({"source_comments": result.get("source_comments")})
```
[the remark this answers is not in the document]
Comment (keshav-space) [reply scanpipe/pipes/symbols.py]: Done

**Pull request merged**